File: CPlusPlus/1.TwoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> hit;
        vector<int> res;
        for (int i = 0; i < nums.size(); ++i) {
            if (hit.find(nums[i]) != hit.end() && i != hit[nums[i]]) {
                res.push_back(hit[nums[i]]);
                res.push_back(i);
                return res;
            }
            hit[target - nums[i]] = i;
        }

    }
};

File: Javascript/1.TwoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */

var twoSum = function(nums, target) {
  let hit = {};
  for (let i = 0; i < nums.length; ++i) {
    if (nums[i] in hit) {
      return [hit[nums[i]], i]
    }
    let hitValue = target - nums[i];
    hit[hitValue] = i;
  }
};